### ISSUE TITLE
Clearing Passphrase from Cache

### ISSUE THREAD

**Issue by romanz** [romanz/trezor-agent#221]
It is not clear how to clear passphrase from `trezor-gpg-agent` cache.

**Comment by romanz** [comment]
The passphrase is cached at `device.trezor.Trezor` class:
https://github.com/romanz/trezor-agent/blob/5dba12f14475b9be3f6cd39276d9138a83c7c515/libagent/device/trezor.py#L49-L76

For simplicity, the caching is done once - so you need to restart `trezor-gpg-agent` process to clear the cache. This can be done using:
```
$ export GNUPGHOME=${HOME}/.gnupg/trezor
$ echo "KILLAGENT" | gpg-connect-agent
```
or:
```
$ killall trezor-gpg-agent
```

**Comment by Iconate** [comment]
Thanks for this information. I can now reset the PIN & Passphrase on demand, as desired. There are still a few weird steps that need to be taken outside of what you have instructed. I have outlined them below

To clear Passphrase `$ echo "KILLAGENT" | gpg-connect-agent` 
To clear PIN `$ trezorctl clear_session`

I can confirm this is the case through using password-store as the only method to trigger both prompts (see point 3). If only 1 (PIN or Passphrase) has been cleared, the password-store commands will trigger the missing PIN or Passphrase before decrypting successfully.

I am running MacOSX 10.13.3, and noticed that I did not have a `gpg-connect-agent` shell command. Homebrew was also saying that `gpg-agent` was already installed. I made a sym link from the Cellar to my /usr/local/bin, which allowed me to clear the passphrase successfully

Note:  `export GNUPGHOME=${HOME}/.gnupg/trezor` is in my `~/.bash_profile`

1. **PIN and Passphrase are not prompt from a gpg command**

When both PIN & Passphrase are cleared, entering

```
echo "123" | gpg --sign | gpg --verify
```

Causes the trezor to flash 3 times (looks like its trying to show the PIN entry prompt), resulting in

```
gpg: Warning: not using '__REDACTED__' as default key: No secret key
gpg: all values passed to '--default-key' ignored
gpg: keydb_search failed: Broken pipe
gpg: no default secret key: Broken pipe
gpg: signing failed: Broken pipe
gpg: verify signatures failed: Unknown system error
```
I think I might be missing something in my gpg.conf.

2. **trezor-gpg unlock only prompts for PIN**
 
After both PIN & Passphrase are cleared, running 
```
$ trezor-gpg unlock
```

Only prompts for the PIN, no passphrase.

3. **The only way I can trigger the passphrase prompt is through password-store**

After clearing PIN & Passphrase,
```
$ pass Test/test.com
``` 
Which triggers the PIN & Passphrase prompts as expected. Then all subsequent GPG calls work as expected
```
$ echo "123" | gpg --sign | gpg --verify
gpg: using "_REDACTED_" as default secret key for signing
gpg: Signature made Fri 16 Mar 12:17:45 2018 EDT
gpg:                using _REDACTED_
gpg:                issuer _REDACTED_
gpg: Good signature from "_REDACTED_" [ultimate]
```

1. Ideally I would like to be able run GPG commands, and receive the PIN & Passphrase prompt, just like password-store does. There might be missing options in my gpg.conf (?).
2. Having only 1 command to clear both PIN & Passphrase would be ideal as well, but I can just set up an alias to run the two commands, not a huge problem.
